src/Tetris/PlayerSkeleton.java

```diff
@@ -1,14 +1,70 @@
 package Tetris;
 
+import java.util.*;
+import java.lang.*;
+
+import Tetris.Util.Tuple;
+import Tetris.Util.Util;
+
 public class PlayerSkeleton {
 
 	//implement this function to have a working system
 	public int pickMove(State s, int[][] legalMoves) {
-		
 		return 0;
 	}
-	
+
 	public static void main(String[] args) {
+		geneticFunction();
+	}
+
+	// Simulates the replacement of the population by its member's descendants
+	public static void geneticFunction() {
+		ArrayList<Heuristic> population = Util.getRandomHeuristics(Constants.NUMBER_OF_HEURISTICS);
+
+		for (int i = 0; i < Constants.NUMBER_OF_GENERATIONS; i++) {
+			HashMap<Heuristic, Integer> averageScores = getPopulationScores(population);
+			population = generateNextGeneration(averageScores);
+		}
+	}
+
+
+	// Creates NUMBER_OF_HEURISTICS new children from the current population and returns this new population.
+	// The probability of two heuristics procreating is proportional to the average score they generated.
+	public static ArrayList<Heuristic> generateNextGeneration(HashMap<Heuristic, Integer> population) {
+		ArrayList<Heuristic> newPopulation = new ArrayList<Heuristic>();
+
+		for (int i = 0; i < population.size(); i++) {
+			Tuple<Heuristic, Integer> mother = randomSelect(population);
```

Comment:
In this line and the next line, can we change the implementation of randomSelect? Because the initialisation to get mother and father should be the same (i.e. getting the interval array), so that can be abstracted out to save some runtime

Reply:
Good observation! Its fixed now :)